iothub/device/src/Transport/AmqpIot/AmqpIotErrorAdapter.cs

```diff
@@ -52,142 +52,113 @@ public static Exception GetExceptionFromOutcome(Outcome outcome)
             return retException;
         }
 
-        public static Exception ToIotHubClientContract(AmqpException amqpException)
+        public static IotHubClientException ToIotHubClientContract(AmqpException amqpException)
         {
             Error error = amqpException.Error;
             AmqpSymbol amqpSymbol = error.Condition;
             string message = error.ToString();
 
+            IotHubClientException retException;
+
             // Generic AMQP error
             if (Equals(AmqpErrorCode.InternalError, amqpSymbol))
             {
-                return new IotHubClientException(message, IotHubClientErrorCode.NetworkErrors, amqpException);
+                retException = new(message, IotHubClientErrorCode.NetworkErrors, amqpException);
```

Comment:
@drwill-ms how do you feel about using ```new()``` in contexts like these where the variable's type isn't immediately obvious?

Should it be
```csharp
retException = new IotHubClientException(...);
```
instead?

Reply:
Yeah, we should basically just use it for class fields where the type must be on the left or for inline init of properties. I agree, in this case it isn't as obvious so we should be explicit.